### ISSUE TITLE
LND crashes: unable to sanity check wallet birthday block

### ISSUE THREAD

**Issue by brandoncurtis** [LN-Zap/zap-desktop#1172]
## Description

After upgrading Zap-Wallet, I'm unable to open my old wallet because Zap crashes while starting up.

## Expected Behavior

Zap-Wallet should update from v0.2.2 to v0.3.1 and run without issue.

## Actual Behavior

After entering a wallet password for a wallet that existed before the update, zap-desktop immediately crashes with this error:

![screenshot from 2018-12-22 21-31-20](https://user-images.githubusercontent.com/2591290/50381076-24f9e880-0632-11e9-84eb-0e9e22d229a7.png)

If I repeatedly open Zap-Wallet, the sync progress continues to advance at the referenced block changes:

![screenshot from 2018-12-22 22-11-38](https://user-images.githubusercontent.com/2591290/50385716-c020ab80-068e-11e9-968c-5a95e4dbd8a4.png)

No matter how many times I reopen Zap-Wallet, the wallet is never successfully loaded.

## Possible Fix

If I create a new wallet in v0.3.1, it works fine.

## Steps To Reproduce

I haven't reproduced this yet.

## Context

Makes it impossible to use an existing wallet when updating from v0.2.2.

I can't revert to v0.2.2 to test if it will work there because "channel db cannot revert to prior version".

The hard drive is not full.  The machine otherwise works fine.

## Your Environment

- Ubuntu 18.04
- Zap-Desktop v0.3.1 built from source
- Replacing v0.2.2 built from source (it worked fine)

**Comment by mrfelton** [comment]
@brandoncurtis how are you connecting to lnd? Are you using Zap's internal lnd or connecting to your own lnd+bitcoin node?

**Comment by mrfelton** [comment]
@brandoncurtis if you quit and restart Zap and login to your wallet again, do you get the same issue?

**Comment by brandoncurtis** [comment]
In this case I'm using Zap's internal LND+neutrino.

On the first sync after the upgrade, I was getting the same-ish behavior after multiple restarts, only the "failed to fetch block hash" height was changing as the sync progressed.  After a dozen restarts the sync had fully completed, and now Zap throws the above error instantly when the wallet is unlocked.

I built the latest version of `lncli` and tried unlocking the wallet manually, and I get the same error.

I thought that maybe I could force some sort of resync and rescan if renamed these files:
- block_headers.bin
- neutrino.db
- reg_filter_headers.bin

But after trying that, it throws the same error immediately after unlock:

```
2018-12-23 14:31:47.739 [INF] LNWL: Opened wallet
2018-12-23 14:31:47.842 [INF] LTND: Primary chain is set to: bitcoin
2018-12-23 14:31:48.220 [INF] BTCN: Loaded 1969 addresses from file '/home/brandon/.config/Zap/lnd/bitcoin/testnet/wallet-1/data/chain/bitcoin/testnet/peers.json'
2018-12-23 14:31:48.220 [INF] BTCN: Waiting for more block headers, then will start cfheaders sync from height 0...
2018-12-23 14:31:48.220 [DBG] CMGR: Attempting to connect to 35.231.86.179:18333 (reqid 2)
2018-12-23 14:31:48.221 [DBG] CMGR: Attempting to connect to 35.196.85.107:18333 (reqid 1)
2018-12-23 14:31:48.308 [INF] CMGR: DNS discovery failed on seed x49.testnet-seed.bitcoin.jonasschnelli.ch: lookup x49.testnet-seed.bitcoin.jonasschnelli.ch on 127.0.0.53:53: no such host
2018-12-23 14:31:48.309 [INF] CMGR: DNS discovery failed on seed x49.seed.tbtc.petertodd.org: lookup x49.seed.tbtc.petertodd.org on 127.0.0.53:53: no such host
2018-12-23 14:31:48.309 [INF] CMGR: DNS discovery failed on seed testnet-seed.bitcoin.schildbach.de: lookup testnet-seed.bitcoin.schildbach.de on 127.0.0.53:53: no such host
2018-12-23 14:31:48.319 [DBG] CMGR: Connected to 35.231.86.179:18333 (reqid 2)
2018-12-23 14:31:48.319 [DBG] CMGR: Connected to 35.196.85.107:18333 (reqid 1)
2018-12-23 14:31:48.319 [DBG] BTCN: Sending version (agent /btcwire:0.5.0/neutrino:0.0.4-beta/, pver 70013, block 0) to 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.319 [DBG] BTCN: Sending version (agent /btcwire:0.5.0/neutrino:0.0.4-beta/, pver 70013, block 0) to 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.419 [DBG] BTCN: Received version (agent /btcwire:0.5.0/btcd:0.12.0/, pver 70013, block 1449158) from 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.419 [DBG] BTCN: Negotiated protocol version 70013 for peer 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.419 [DBG] BTCN: Received version (agent /btcwire:0.5.0/btcd:0.12.0/, pver 70013, block 1449158) from 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.419 [DBG] BTCN: Negotiated protocol version 70013 for peer 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.419 [DBG] BTCN: Added time sample of 0s (total: 1)
2018-12-23 14:31:48.419 [DBG] BTCN: Added time sample of 0s (total: 2)
2018-12-23 14:31:48.419 [DBG] BTCN: Connected to 35.231.86.179:18333
2018-12-23 14:31:48.419 [DBG] BTCN: Connected to 35.196.85.107:18333
2018-12-23 14:31:48.419 [INF] BTCN: New valid peer 35.231.86.179:18333 (outbound) (/btcwire:0.5.0/btcd:0.12.0/)
2018-12-23 14:31:48.420 [DBG] BTCN: Received verack from 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.420 [DBG] BTCN: Received verack from 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.420 [INF] BTCN: Syncing to block height 1449158 from peer 35.231.86.179:18333
2018-12-23 14:31:48.420 [DBG] BTCN: Sending verack to 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.420 [DBG] BTCN: Sending verack to 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.420 [INF] BTCN: Downloading headers for blocks 1 to 546 from peer 35.231.86.179:18333
2018-12-23 14:31:48.420 [INF] BTCN: New valid peer 35.196.85.107:18333 (outbound) (/btcwire:0.5.0/btcd:0.12.0/)
2018-12-23 14:31:48.419 [DBG] BTCN: New peer 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.420 [DBG] BTCN: New peer 35.196.85.107:18333 (outbound)
2018-12-23 14:31:48.420 [DBG] BTCN: Sending getheaders (locator 000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943, stop 000000002a936ca763904c3c35fce2f3556c559c0214345d31b1bcebf76acb70) to 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.521 [INF] CMGR: 3 addresses found from DNS seed testnet-seed.bluematt.me
2018-12-23 14:31:48.816 [DBG] BTCN: Received headers (num 546) from 35.231.86.179:18333 (outbound)
2018-12-23 14:31:48.820 [INF] BTCN: Verified downloaded block header against checkpoint at height 546/hash 000000002a936ca763904c3c35fce2f3556c559c0214345d31b1bcebf76acb70
2018-12-23 14:31:48.916 [DBG] BTCN: Sending getheaders (locator 000000002a936ca763904c3c35fce2f3556c559c0214345d31b1bcebf76acb70, stop 00000000009e2958c15ff9290d571bf9459e93b19765c6801ddeccadbb160a1e) to 35.231.86.179:18333 (outbound)
2018-12-23 14:31:49.039 [INF] LNWL: The wallet has been unlocked without a time limit
2018-12-23 14:31:49.039 [INF] LTND: LightningWallet opened
2018-12-23 14:31:49.039 [DBG] LNWL: Starting sanity check for the wallet's birthday block from: height=410945, hash=000000000001a834a748448dc748e45ead0caeaacc65994c86c2b6c582f270b4
2018-12-23 14:31:49.039 [ERR] LNWL: unable to sanity check wallet birthday block: unable to get header for block hash 000000000001a834a748448dc748e45ead0caeaacc65994c86c2b6c582f270b4: target hash not found in index
```

**Comment by mrfelton** [comment]
@brandoncurtis I'm not sure what the cause of this is, and it's not something that I have seen before. But, it does appear that the error is coming directly from Lnd, which is crashing with a panic when trying to open your older wallet in the most recent version of Lnd (we ship 0.5.1 inZap 0.3.1).

You might want to create an issue over in the Lnd project and see if they can help you diagnose the issue. It sounds like a problem with their wallet migration process for older wallets.

**Comment by mrfelton** [comment]
@bg002h your issues sounds like something different from the issue reported here. If you could please create a new issue [here](https://github.com/LN-Zap/zap-desktop/issues/new/choose) and provide as much detail as you can about your setup and your problem we can try to help you diagnose.

**Comment by brandoncurtis** [comment]
Someone else reported something similar in LND: https://github.com/lightningnetwork/lnd/issues/2403

The only PR in LND that I can find that seems to be related appears to have been included in v0.5.1: https://github.com/lightningnetwork/lnd/commit/d3b44542d395fe31ee4a9feb18b935266b42a6ab

I'm going to close this issue and someone else can reopen it if this problem is spotted again.
